dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImplTest.java

```diff
@@ -645,271 +659,335 @@ public void testSearch () throws Exception {
+
+        Contentlet content = new ContentletDataGen(type.id()).setProperty("title",
+                        "contentTest " + System.currentTimeMillis())
+                .setProperty("body", testBody).next();
+
+        String title = "version1";
+        content.setStringProperty("title", title);
+
+        content.setIndexPolicy(IndexPolicy.FORCE);
+
+        // check in the content
+        content = contentletAPI.checkin(content, user, false);
+
+        assertNotNull(content.getIdentifier());
+        assertTrue(content.isWorking());
+        assertFalse(content.isLive());
+        assertEquals(content.getTitle(), title);
+        // publish the content
+        content.setIndexPolicy(IndexPolicy.FORCE);
+        content.setBoolProperty(Contentlet.DISABLE_WORKFLOW, true);
+        contentletAPI.publish(content, user, false);
+        String liveInode = content.getInode();
+        assertTrue(content.isLive());
+        content.setInode(null);
+        title = "version2";
+        content.setStringProperty("title", title);
+        content.setIndexPolicy(IndexPolicy.FORCE);
+        content = contentletAPI.checkin(content, user, false);
+        assertNotNull(content.getIdentifier());
+        assertTrue(content.isWorking());
+        assertFalse(content.isLive());
+        assertTrue(content.hasLiveVersion());
+        String workingInode = content.getInode();
+        final String identifier = content.getIdentifier();
+
+        List<ContentletSearch> liveSearchResults = pollSearchIndex(1, identifier, null, true, 1, 0,
+                MOD_DATA_SORT, user);
+        assert (liveSearchResults.get(0).getInode().equals(liveInode));
+        List<ContentletSearch> workingSearchResults = pollSearchIndex(1, identifier, null, false, 1,
+                0, MOD_DATA_SORT, user);
+        assert (workingSearchResults.get(0).getInode().equals(workingInode));
+        assert (!workingSearchResults.get(0).getInode().equals(liveInode));
+
+        indexAPI.removeContentFromIndex(content);
+
+        pollSearchIndex(0, identifier, null, true, 1, 0, MOD_DATA_SORT, user);
+        pollSearchIndex(0, identifier, null, false, 1, 0, MOD_DATA_SORT, user);
+
+        Contentlet finalContent = content;
+        LocalTransaction.wrap(() -> {
+            APILocator.getReindexQueueAPI().addContentletReindex(finalContent);
+            ReindexThread.startThread();
+        });
+
+        // wait for empty reindex queue
+
+        waitForEmptyQueue();
+
+        liveSearchResults = pollSearchIndex(1, identifier, null, true, 1, 0, MOD_DATA_SORT, user);
+        workingSearchResults = pollSearchIndex(1, identifier, null, false, 1, 0, MOD_DATA_SORT,
+                user);
+
+        assert (liveSearchResults.get(0).getInode().equals(liveInode));
+
+        assert (workingSearchResults.get(0).getInode().equals(workingInode));
+        assert (!workingSearchResults.get(0).getInode().equals(liveInode));
+
+    }
+
+    public void waitForEmptyQueue() {
```

Comment:
why these methods `waitForEmptyQueue` , `pollSearchIndex` and `searchIndex` are public?

Reply:
Agree,  have updated